docs/blockentities/ber.md

```diff
@@ -1,44 +1,119 @@
+            context -> new MyBlockEntityRenderer()
+    );
+}
+```
 
-```java
-public class MyBlockEntityRenderer implements BlockEntityRenderer {
+## `BlockEntityWithoutLevelRenderer`
 
-    public MyBlockEntityRenderer(BlockEntityRendererProvider.Context ctx) {
-        // Do things here
+`BlockEntityWithoutLevelRenderer`, colloquially known as BEWLR, is an adaptation of the regular `BlockEntityRenderer` for special [item] rendering (hence "without level", as items do not have level context). Its overall purpose is the same: do special rendering for cases where static models aren't enough.
+
+To add a BEWLR, create a class that extends `BlockEntityWithoutLevelRenderer` and overrides `#renderByItem`. It also requires some additional constructor setup:
+
+```java
+public class MyBlockEntityWithoutLevelRenderer extends BlockEntityWithoutLevelRenderer {
+    // We need some boilerplate in the constructor, telling the superclass where to find the central block entity and entity renderers.
+    public MyBlockEntityWithoutLevelRenderer() {
+        super(Minecraft.getInstance().getBlockEntityRenderDispatcher(), Minecraft.getInstance().getEntityModels());
+    }
+    
+    @Override
+    public void renderByItem(ItemStack stack, ItemDisplayContext transform, PoseStack poseStack, MultiBufferSource bufferSource, int packedLight, int packedOverlay) {
+        // Do the rendering here.
     }
+}
+```
+
+Keep in mind that, like with BERs, there is only one instance of your BEWLR. Stack-specific properties should therefore be stored in the stack, not the BEWLR.
 
-    // Implement #render method here and any other logic
+Unlike BERs, we do not register BEWLRs directly. Instead, we register an instance of `IClientItemExtensions` to the `RegisterClientExtensionsEvent`. `IClientItemExtensions` is an interface that allows us to specify a number of rendering-related behaviors on items, such as (but not limited to) a BEWLR. As such, our implementation of that interface could look like so:
+
+```java
+public class MyClientItemExtensions implements IClientItemExtensions {
+    // Cache our BEWLR in a field.
+    private final MyBlockEntityWithoutLevelRenderer myBEWLR = new MyBlockEntityWithoutLevelRenderer();
+
+    // Return our BEWLR here.
+    @Override
+    public BlockEntityWithoutLevelRenderer getCustomRenderer() {
+        return myBEWLR;
+    }
 }
+```
+
+And then, we can register our `IClientItemExtensions` to the event:
 
-// In another class using some method to listen to this event
+```java
 @SubscribeEvent
-public static void registerRenderers(EntityRenderersEvent.RegisterRenderers event) {
-    event.registerBlockEntityRenderer(MyBlockEntityTypes.MYBE.get(), MyBlockEntityRenderer::new);
+public static void registerClientExtensions(RegisterClientExtensionsEvent event) {
+    event.registerItem(
+            // The only instance of our IClientItemExtensions, and as such, the only instance of our BEWLR.
+            new MyClientItemExtensions(),
```

Comment:
I would add a note that there should only ever be one instance of an extension subclass, as it is essentially treated as a singleton itself where stack methods are routed through.